src/services/streams.ts

```diff
@@ -169,60 +152,38 @@ export const getPagedStreams = async (
     lastId?: string,
     owner?: string,
     search?: string,
-    orderBy?: TheGraphOrderBy,
-    orderDirection?: TheGraphOrderDirection,
+    orderBy?: Stream_OrderBy,
+    orderDirection?: OrderDirection,
 ): Promise<TheGraphStreamResult> => {
-    const theGraphUrl = getGraphUrl()
+    const orderOperator = orderDirection === OrderDirection.Asc ? 'gt' : 'lt'
 
-    // NOTE: Stream name fulltext search is done through subentity "permissions" because we cannot
-    // use "id_contains" in query as it's not technically stored as a string on The Graph.
-    const searchFilter =
-        search != null && search.length > 0 ? `stream_contains_nocase: "${search}"` : null
-    const ownerFilter =
-        owner != null ? `userAddress: "${owner.toLowerCase()}", canGrant: true` : null
-    const allPermissionFilters = [ownerFilter, searchFilter]
-        .filter((filter) => !!filter)
-        .join(',')
-    const permissionFilter =
-        allPermissionFilters.length > 0 && `permissions_: { ${allPermissionFilters} }`
-    const comparisonOperator = orderDirection === TheGraphOrderDirection.Asc ? 'gt' : 'lt'
-    const cursorFilter = lastId != null ? `id_${comparisonOperator}: "${lastId}"` : null
-    const allFilters = [cursorFilter, permissionFilter]
-        .filter((filter) => !!filter)
-        .join(',')
+    let where: Stream_Filter = {
+        permissions_: {
+            stream_contains_nocase: search,
+            userAddress: owner,
+        },
+        [`id_${orderOperator}`]: lastId,
+    }
 
-    const result = await post({
-        url: theGraphUrl,
-        data: {
-            query: `
-                {
-                    streams(
-                        first: ${first + 1},
-                        orderBy: ${orderBy?.toString() ?? TheGraphOrderBy.Id.toString()},
-                        orderDirection: ${
-                            orderDirection?.toString() ??
-                            TheGraphOrderDirection.Asc.toString()
-                        },
-                        ${allFilters != null ? `where: { ${allFilters} }` : ''},
-                    ) {
-                        id
-                        metadata
-                        permissions {
-                            userAddress
-                            canEdit
-                            canGrant
-                            canDelete
-                            subscribeExpiration
-                            publishExpiration
-                        }
-                    }
-                }
-            `,
+    where.permissions_ = _.omitBy(where.permissions_, _.isEmpty)
+    where = _.omitBy(where, _.isEmpty)
+
+    const {
+        data: { streams },
+    } = await getGraphClient().query<GetPagedStreamsQuery, GetPagedStreamsQueryVariables>(
+        {
+            query: GetPagedStreamsDocument,
+            variables: {
+                first: first + 1,
+                orderBy: orderBy ?? Stream_OrderBy.Id,
```

Comment:
Any advantage of using `??` here?

Reply:
yeah, those could be default param values